### DISCUSSION TITLE
Remove 'Humidify' selector from Dehumidifier accessory

### DISCUSSION THREAD

**Discussion by bramboe** [HomeSpan/HomeSpan#361] [General]
Hi! 

I'm playing around by creating an accessory. I'm focussing on creating a Dehumidifier accessory. From the Homespan setup everything seems to work!  (I also hooked up a DHT22 to it as a test :) )

I noticed the switch keep showing 'Humidify', which does not exist on a Dehumidifier. Any Idea on how to remove this selector from the options?

**Comment by HomeSpan** [comment]
You should be able to exclude choices for a target state using the method SpanCharacteristic *setValidValues().  See the API reference for details and an example.

**Comment by bramboe** [comment]
Thank you!

This is the code that worked for me:
`    (new Characteristic::TargetHumidifierDehumidifierState())->setValidValues(3,0,2,3);            
`

**Reply by dgalue** [reply]
Hi,

Any change to share your code?
